magefile.go

```diff
@@ -34,13 +34,42 @@ var releases = map[string]string{
 
 type Build mg.Namespace
 
+// helper function to populate build sources with golang source files
+func goSources() []string {
+	sources := make([]string, 0)
+
+	cwd, _ := os.Getwd()
+	if err := filepath.Walk(cwd, func(path string, f os.FileInfo, err error) error {
+		if err != nil || f.IsDir() {
+			return err
+		}
+
+		// match all golang source files (except for the magefile)
+		if ok, _ := filepath.Match("*.go", f.Name()); ok && f.Name() != "magefile.go" {
```

Comment:
This works, but it may be better to match against the build target in the special comment at the top of the file, that way if this mage file is ever expanded into multiple files it matches all of them.  Not sure if my suggestion is feasible though?

Reply:
The consequences of not having this check are so minor, I'd rather just remove it than extend it into something more verbose and complex for little or no tangible benefit.

The only consequence is that the build will be run again (with the same output) if only the magefile has changed.